### PR TITLE
[Opaque Pointer] Remove non opaque pointer tests and useless opaque-pointer options

### PR DESCRIPTION
**Pull request by jsji**
This is in preparation for setting -DINTEL_SYCL_OPAQUEPOINTER_READY=1
- Remove no opaque CodeGenSYCL tests
- Remove no-opaque-pointer
- Remove clang opaque pointer

### CODE REVIEW

**Review by bader — APPROVED** [review]

**Review by mdtoguchi — APPROVED** [review]

File: clang/test/CodeGenSYCL/annotations-field-no-opaque.cpp
[the remark this answers is not in the document]
Comment (Fznamznon) [reply clang/test/CodeGenSYCL/annotations-field-no-opaque.cpp:1]: Why it shouldn't be not okay to remove? If the target  is to move to opaque pointers, then this test in not necessary anymore. https://github.com/intel/llvm/commit/8e13852c23aff6b7c0ba790df2bcc4dfe83d09e6 is not SYCL specific (however its "tag" says the opposite) but it exists only in intel/llvm.

**Review by dm-vodopyanov — APPROVED** [review]
`sycl/test/` changes LGTM

**Comment by jsji** [comment]
Thanks all for the prompt review and approval. :)

**Review by Fznamznon — APPROVED** [review]
Though, I didn't check every test and that we don't remove any options that exist in the community code

**Review by smanna12 — APPROVED** [review]

**Comment by jsji** [comment]
@intel/llvm-gatekeepers  Can we get this merged? Thanks!

**Comment by jsji** [comment]
> Though, I didn't check every test and that we don't remove any options that exist in the community code

Thanks. This PR only removes opaque-pointer related options , the opaque pointer related options were already removed in community , there shouldn't be any test still using them.

**Pull request merged**